Elements/UIResText.cs

```diff
@@ -39,14 +39,68 @@ public UIResText(string caption, Point position, float scale, Color color, Font
         /// <param name="str"></param>
         public static void AddLongString(string str)
         {
-            const int strLen = 99;
-            for (int i = 0; i < str.Length; i += strLen)
+            var utf8ByteCount = System.Text.Encoding.UTF8.GetByteCount(str);
+
+            if (utf8ByteCount == str.Length)
+            {
+                AddLongStringForAscii(str);
+            }
+            else
+            {
+                AddLongStringForUtf8(str);
+            }
+        }
+
+        private static void AddLongStringForAscii(string input)
+        {
+            const int maxByteLengthPerString = 99;
+
+            for (int i = 0; i < input.Length; i += maxByteLengthPerString)
             {
-                string substr = str.Substring(i, Math.Min(strLen, str.Length - i));
+                string substr = (input.Substring(i * maxByteLengthPerString, Math.Min(maxByteLengthPerString, input.Length - i * maxByteLengthPerString)));
```

Comment:
I don't think that's right. Either do that and increment i by 1, or do it as it was before